test/new-e2e/tests/agent-shared-components/hostname/imdsv2_transition_common_test.go

```diff
@@ -0,0 +1,197 @@
+	statusPayload := v.Env().Agent.Client.Status(statusArgs)
+
+	var data map[string]interface{}
+	if err := json.Unmarshal([]byte(statusPayload.Content), &data); err != nil {
+		v.T().Fatal(err)
+	}
+
+	metaMap, ok := data["metadata"].(map[string]interface{})["meta"].(map[string]interface{})
+	if !ok {
+		v.T().Fatal()
+	}
+
+	var hostnameMeta Meta
+	hostnameMeta.Hostname, _ = metaMap["hostname"].(string)
+	hostnameMeta.LegacyResolutionHostname, _ = metaMap["legacy-resolution-hostname"].(string)
+
+	v.hostnameMetadata = hostnameMeta
+}
+
+// retrieveMetadata retrieves the instance ID and OS hostname from the EC2 metadata
+func retrieveMetadata(v *baseHostnameSuite) (string, string) {
+	metadata := client.NewEC2Metadata(v.T(), v.Env().RemoteHost.Host, v.Env().RemoteHost.OSFamily)
+	instanceID := metadata.Get("instance-id")
+	osHostname := strings.Split(metadata.Get("hostname"), ".")[0]
+	return instanceID, osHostname
+}
+
+// runHostnameTest runs the hostname test with the given parameters
+func runHostnameTest(v *baseHostnameSuite, instanceOpts []awshost.ProvisionerOption, imdsV1Enable bool, tt struct {
+	name                     string
+	ec2PreferIMDSv2          bool
+	legacyResolutionHostname bool
+}) {
+	v.T().Run(tt.name, func(t *testing.T) {
+		v.UpdateEnv(awshost.ProvisionerNoFakeIntake(instanceOpts...))
+
+		requestAgentHostnameMetadataPayload(v)
+
+		instanceID, osHostname := retrieveMetadata(v)
+
+		assert.NotEmpty(t, v.hostnameMetadata.Hostname)
+
+		// legacy-resolution-hostname should only be set when IMDSv1 is disabled, ec2_prefer_imdsv2 is disabled and legacy-resolution-hostname is enabled
+		if tt.legacyResolutionHostname && !imdsV1Enable && !tt.ec2PreferIMDSv2 {
+			assert.NotEmpty(t, v.hostnameMetadata.LegacyResolutionHostname)
+			assert.NotEqual(t, v.hostnameMetadata.Hostname, v.hostnameMetadata.LegacyResolutionHostname)
+			assert.Equal(t, v.hostnameMetadata.LegacyResolutionHostname, osHostname)
+		} else {
+			assert.Empty(t, v.hostnameMetadata.LegacyResolutionHostname)
+		}
+
+		expectedHostname := instanceID
+
+		// The hostname should be the OS hostname when IMDS is disabled
+		if !imdsV1Enable && !tt.ec2PreferIMDSv2 && !tt.legacyResolutionHostname {
+			expectedHostname = osHostname
+		}
+
+		assert.Equal(t, expectedHostname, v.hostnameMetadata.Hostname)
+	})
+}
+
+// TestWithIMDSv1 tests the default hostname resolution for ec2 instances when IMDSv1 is enabled which means the hostname should always be the instance ID
+func (v *baseHostnameSuite) TestWithIMDSv1() {
+	tests := []struct {
+		name                     string
+		ec2PreferIMDSv2          bool
+		legacyResolutionHostname bool
+	}{
+		// IMDSv1 is enable, transition to IMDSv2 should not change any behavior, legacy-resolution-hostname should be ignored and not set
+		{
+			name:                     "WithIMDSv1_IMDSv2Transition_NoLegacy",
+			ec2PreferIMDSv2:          false,
+			legacyResolutionHostname: true,
+		},
```

Comment:
Did you consider adding the expected value for `hostname` and `LegacyResolutionHostname` in this struct?